src/react-components/ui-root.js

```diff
@@ -195,6 +204,37 @@ class UIRoot extends Component {
     this.props.scene.addEventListener("share_video_disabled", this.onShareVideoDisabled);
     this.props.scene.addEventListener("exit", this.exit);
     const scene = this.props.scene;
+
+    let preEntryHistoryLength = 0;
+
+    // If we landed on the page with a path in the middle of the entry flow (eg we refreshed the
+    // page on the audio setup dialog) then reset the history entry to /.
+    //
+    // Note this isn't perfect, if we refresh the page mid-entry flow and then hit back, we end
+    // up in a bad state unless we were on the first step. But this seems reasonable enough for now.
+    if (ENTRY_FLOW_ROUTES.find(x => x === this.props.history.location.pathname)) {
+      this.props.history.replace("/");
+    }
+
+    // Hacky technique to skip over the entry flow history entries when we've entered the room and
+    // hit the back button.
+    //
+    // This makes it so if we are in the room and hit back in the browser, we go to the URL
+    // the browser was on before the entry flow, not back into the entry flow.
+    this.props.history.listen((newLocation, action) => {
```

Comment:
Hmm this seems super fishy. The entered state should just be a route, and that route should `replace` all the entry flow routes in the history stack.